src/app/guards/load-streams.guard.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree } from '@angular/router';
+import {combineLatest, Observable, of} from 'rxjs';
+import {Store} from '@ngrx/store';
+import {loadStreams} from '../store/stream/stream.actions';
+import {filter, take, tap, switchMap} from 'rxjs/operators';
+import {getIsStreamLoaded, getIsStreamLoading} from '../store/stream/stream.selectors';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoadStreamsGuard implements CanActivate {
+  constructor(private store: Store) {}
+
+  canActivate(
```

Comment:
This is an auto generated method but since you are not using any of those arguments ;) :
```suggestion
  canActivate(): Observable<boolean> {
```